pkg/controller/releasemanager/plan/syncApp.go

```diff
@@ -97,106 +96,152 @@ func (p *SyncApp) serviceHost() string {
+}
 
-	portNumber := uint32(port.Port)
-	gateways := []string{"mesh"}
-	hosts := port.Hosts
-
-	switch port.Mode {
-	case picchuv1alpha1.PortPublic:
-		if p.PublicGateway == "" {
-			log.Info("publicGateway undefined in Cluster for port", "port", port)
-			return matches
-		}
-		gateways = []string{p.PublicGateway}
-		portNumber = uint32(port.IngressPort)
-	case picchuv1alpha1.PortPrivate:
-		if p.PrivateGateway == "" {
-			log.Info("privateGateway undefined in Cluster for port", "port", port)
-			return matches
-		}
-		gateways = []string{p.PrivateGateway}
-
-		defaultDomainsMap := map[string]bool{}
-		for _, defaultDomain := range p.DefaultDomains {
-			defaultDomainsMap[defaultDomain] = true
-			if clusterName != "" {
-				clusterDomain := fmt.Sprintf("%s.cluster.%s", clusterName, defaultDomain)
-				defaultDomainsMap[clusterDomain] = true
-			}
-		}
-		for domain := range defaultDomainsMap {
-			hosts = append(hosts, fmt.Sprintf("%s.%s", p.Namespace, domain))
-			if port.Name != "" {
-				hosts = append(hosts, fmt.Sprintf("%s-%s.%s", p.Namespace, port.Name, domain))
-			}
-		}
+func (p *SyncApp) privateHosts(port picchuv1alpha1.PortInfo, options plan.Options) []string {
+	return p.ingressHosts(picchuv1alpha1.PortPrivate, port, options.DefaultDomains.Private)
+}
 
-		portNumber = uint32(port.IngressPort)
+func (p *SyncApp) authorityMatch(hosts []string) *istio.StringMatch {
+	sort.Strings(hosts)
+	regex := fmt.Sprintf("^(%s)", strings.Join(hosts, "|"))
+	return &istio.StringMatch{
+		MatchType: &istio.StringMatch_Regex{Regex: regex},
 	}
+}
 
-	for _, host := range hosts {
-		matches = append(matches, &istio.HTTPMatchRequest{
-			Authority: &istio.StringMatch{MatchType: &istio.StringMatch_Prefix{Prefix: host}},
-			Port:      portNumber,
-			Gateways:  gateways,
-			Uri:       &istio.StringMatch{MatchType: &istio.StringMatch_Prefix{Prefix: "/"}},
-		})
-		if port.IngressPort == 443 && port.HttpsRedirect {
-			matches = append(matches, &istio.HTTPMatchRequest{
-				Authority: &istio.StringMatch{MatchType: &istio.StringMatch_Prefix{Prefix: host}},
-				Port:      uint32(80),
-				Gateways:  gateways,
-				Uri:       &istio.StringMatch{MatchType: &istio.StringMatch_Prefix{Prefix: "/"}},
-			})
+func (p *SyncApp) publicAuthorityMatch(port picchuv1alpha1.PortInfo, options plan.Options) *istio.StringMatch {
+	return p.authorityMatch(p.publicHosts(port, options))
+}
+
+func (p *SyncApp) privateAuthorityMatch(port picchuv1alpha1.PortInfo, options plan.Options) *istio.StringMatch {
+	return p.authorityMatch(p.privateHosts(port, options))
+}
+
+func (p *SyncApp) ingressHosts(
+	mode picchuv1alpha1.PortMode,
+	port picchuv1alpha1.PortInfo,
+	defaultDomains []string,
+) []string {
+	hostMap := map[string]bool{}
+	defaultDomainsMap := map[string]bool{}
+	for _, defaultDomain := range defaultDomains {
+		defaultDomainsMap[defaultDomain] = true
+	}
+	for domain := range defaultDomainsMap {
+		hostMap[fmt.Sprintf("%s.%s", p.Namespace, domain)] = true
```

Comment:
kbfd expects this to only be the full namespace if the app’s target is different from the fleet name:

```
lite.production.medm.io
lite-staging.internal.medm.io
```